### PR TITLE
refactor(function): getNewValue Function

### PR DESCRIPTION
**Pull request by souravdasslg**
1. Create interface NewValueInterface
2. Update function calls
3. Update tests

#4996

### CODE REVIEW

File: lib/versioning/loose/generic.ts [hunk 1]
```diff
@@ -191,10 +192,11 @@ export abstract class GenericVersioningApi<
...
-    _fromVersion: string,
-    toVersion: string
+    { toVersion }: NewValueConfig = {
+      currentValue: null,
+      rangeStrategy: null,
+      toVersion: arguments[0],
+    }
   ): string {
     return toVersion;
   }
```
Comment (souravdasslg) [inline lib/versioning/loose/generic.ts]: @rarkins I don't think, this as a good approach. Any idea to make it better?

Comment (rarkins) [reply lib/versioning/loose/generic.ts]: Can we un-expand the params and just make it like `getNewValue(newValueConfig: NewValueConfig)` and then `return newValueConfig.toVersion` ?

Comment (souravdasslg) [reply lib/versioning/loose/generic.ts]: > That looks very strange to me. Why we need `arguments[0]`?

If the value is falsy then it cannot be destructured. The test was expecting the same value as it is expecting.

Comment (souravdasslg) [reply lib/versioning/loose/generic.ts]: > Can we un-expand the params and just make it like `getNewValue(newValueConfig: NewValueConfig)` and then `return newValueConfig.toVersion` ?

What if it is a falsy value? `return` will break there.

Comment (ViceIce) [reply lib/versioning/loose/generic.ts]: then we should simply use `return arg?.toVersion`  <-- New typescript null check

**Comment by rarkins** [comment]
By the way, I think this PR is a great example of where passing objects as params results in much more readable and maintainable code (even without TS)

File: lib/versioning/loose/generic.ts [hunk 2]
```diff
@@ -81,10 +81,11 @@ export const comparer = (
...
   function getNewValue(
-    _currentValue: string,
-    _rangeStrategy: RangeStrategy,
-    _fromVersion: string,
-    toVersion: string
+    { toVersion }: NewValueConfig = {
+      currentValue: null,
+      rangeStrategy: null,
+      toVersion: arguments[0],
```
Comment (rarkins) [inline lib/versioning/loose/generic.ts]: @souravdasslg can this be refactored too? I find this syntax really confusing

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/versioning/loose/generic.ts (hunk 2 above)
Comment (souravdasslg) [reply lib/versioning/loose/generic.ts]: Sure. Ill do that.

**Review by rarkins — APPROVED** [review]

**Pull request merged**